core/blockchain_eip3529_test.go

```diff
@@ -0,0 +1,165 @@
+package core
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus"
+	"github.com/ethereum/go-ethereum/consensus/ethash"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+func postLondonConfig() *params.ChainConfig {
+	config := *params.TestChainConfig
+	config.LondonBlock = big.NewInt(0)
+	return &config
+}
+
+func preLondonConfig() *params.ChainConfig {
+	config := *params.TestChainConfig
+	config.LondonBlock = nil
+	return &config
+}
+
+func TestSelfDestructGasPreEIP3529(t *testing.T) {
+	bytecode := []byte{
+		byte(vm.PC),
+		byte(vm.SELFDESTRUCT),
+	}
+
+	// Expected gas is (intrinsic +  pc + cold load (due to legacy tx)  + selfdestruct cost ) / 2
+	// The refund of 24000 gas (i.e. params.SelfdestructRefundGas) is not applied since refunds pre-EIP3529 are
+	// capped to half of the transaction's gas.
+	expectedGasUsed := (params.TxGas + vm.GasQuickStep + params.ColdAccountAccessCostEIP2929 + params.SelfdestructGasEIP150) / 2
+	testGasUsage(t, preLondonConfig(), ethash.NewFaker(), bytecode, nil, 60_000, expectedGasUsed)
```

Comment:
could be changed once Berlin is enabled?

Reply:
What do you mean?  Berlin is enabled in `TestchainConfig`.

Reply:
I thought `PreEIP3529` would be `Istanbul `, it is ok if the Berlin is enabled in `TestchainConfig `